### PR TITLE
Change logic for defining owner of namespaced field for __typename querying

### PR DESCRIPTION
**Pull request by felipe-gdr**

### CODE REVIEW

File: test/src/test/resources/fixtures/namespaced/typename-is-sent-to-owning-service-even-when-no-fields-are-queried.yml [hunk 1]
```diff
@@ -0,0 +1,199 @@
+name: typename is sent to owning service even when no fields are queried
```
Comment (felipe-gdr) [inline test/src/test/resources/fixtures/namespaced/typename-is-sent-to-owning-service-even-when-no-fields-are-queried.yml:1]: Added a test where the query doesn't contain any fields from the service that owns the namespaced field

Comment (temaEmelyan) [reply test/src/test/resources/fixtures/namespaced/typename-is-sent-to-owning-service-even-when-no-fields-are-queried.yml:1]: I think would be good to have a test with the same scenario that surfaced the bug. The namespaced type is extended in the same service where it's defined. What do you think?

Comment (gnawf) [reply test/src/test/resources/fixtures/namespaced/typename-is-sent-to-owning-service-even-when-no-fields-are-queried.yml:1]: Yep having both this test and the one @temaEmelyan suggested would be great.

**Review by temaEmelyan — APPROVED** [review]

File: engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java [hunk 2]
```diff
@@ -258,11 +258,11 @@ private static boolean fieldMatchesDefinition(GraphQLFieldDefinition graphQLFiel
     }
 
     private static boolean serviceOwnsNamespacedField(GraphQLObjectType namespacedObjectType, Service service) {
-        final boolean serviceExtendsNamespaceType = service.getDefinitionRegistry().getDefinitions(ObjectTypeExtensionDefinition.class)
+        return service.getDefinitionRegistry().getDefinitions(ObjectTypeDefinition.class)
                 .stream()
+                // the type can't be an extension in the owning service
+                .filter(objectTypeDef -> !(objectTypeDef instanceof ObjectTypeExtensionDefinition))
```
Comment (gnawf) [inline engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java:264]: Ah, I guess that's why you did the no object ext instead.

**Review by gnawf — APPROVED** [review]

File: engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java (hunk 2 above)
Comment (felipe-gdr) [reply engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java:264]: Yep

**Pull request merged**